### DISCUSSION TITLE
Opt In Typography?

### DISCUSSION THREAD

**Discussion by devtrix** [skeletonlabs/skeleton#1998] [General]
I am not getting this whole thing.  When I declare an h1 or an h2 I expect that to be so instead of adding the respective classes.  Is the a setting to respect the elements?  I hope I am making sense with this question.

**Comment by endigo9740** [comment]
This was introduced back in v1.5.1. Here's the release notes with details:
- https://github.com/skeletonlabs/skeleton/discussions/1457

> This provides a number of benefits: including making it easy to avoid unwanted styling, allowing for local overrides inline with utility classes, as well as providing simpler selectors for global style overrides in your apps.

**Reply by devtrix** [reply]
I am sure it was in that release.  An Html tag is supposed to mean what it represents just rice is rice, wheat is wheat, a horse is a horse. SMH.  I just don't get this logic.  Its onething to enhance the tag but its another to nullify what it means and force the class on it to indicate what it is.  But hey, if one doesnt like it, move on right.  Just saying that its ridiculous.  But thanks for your response.  Cheers.

**Reply by endigo9740** [reply]
The rude remark isn't really needed. Especially when you're seemingly failing to take into account accessibility. If you're new to web accessibility when it comes to headings you can learn more here:

https://www.w3.org/WAI/tutorials/page-structure/headings/

In a nutshell the heading markup should decent as you indent, which is beneficial for screen readers, but the visual presentation of this might not always match in all scenarios. So separating markup and styling is actually a very effective manner to handle this. 

Furthermore, we tried it the other way and it had a number of issues for a lot of our users. Once we switched to the opt-in approach using utility classes (something very familiar to folks using Tailwind) those issues went away. Folks have options now rather than being forced into a single scenario that doesn't work for all.

**Reply by Sarenor** [reply]
@devtrix also, if you like it, you can just
```css
h1 {
   @apply .h1
}
```
in your app.postcss and get the fell of old typography back.

The big problem with opt-out is that people want to have unstyled headings and links and the css selectors to get them unstyled again after styling them by default are a pain. Especially when some styles should stay and need to be re-applied by the user.
